snake-pygame/snake_game.py

```diff
@@ -62,13 +62,25 @@ def play_step(self):
                 quit()
             if event.type == pygame.KEYDOWN:
                 if event.key == pygame.K_LEFT:
-                    self.direction = Direction.LEFT
+                    if self.direction == Direction.RIGHT:
+                        self.direction == Direction.RIGHT
```

Comment:
=, not ==. Also, this is a little bit too much unnecessary code. You can simply use if self.direction != Direction.RIGHT